src/app.js

```diff
@@ -1,20 +1,64 @@
-const express = require('express');
-const app = express();
+require('dotenv').config()
+const express = require('express')
+const app = express()
+const morgan = require('morgan')
+const bodyParser = require('body-parser')
 const cors = require('cors')
-const favicon = require('express-favicon');
-const logger = require('morgan');
+const favicon = require('express-favicon')
+const mainRouter = require('./routes/mainRouter.js')
+const helmet = require('helmet')
+const session = require('express-session')
+const passport = require('passport')
+const mongoose = require('mongoose')
+const asyncErrors = require('express-async-errors')
+const cookieParser = require('cookie-parser')
+const rateLimit = require('express-rate-limit')
+const xss = require('xss')
+const mongoSanitize = require('mongo-sanitize')
+const limiter = rateLimit({
+  windowMs: 15 * 60 * 1000, // 15 minutes
+  max: 60 // limit each IP to 100 requests per windowMs
+})
 
-const mainRouter = require('./routes/mainRouter.js');
-
-// middleware
-app.use(cors());
-app.use(express.json());
-app.use(express.urlencoded({ extended: false }));
-app.use(logger('dev'));
+// middleware setup
+app.use(cors())
```

Comment:
For now, leaving this empty will work fine with the frontend since it's not trying to pass any cookies in the request yet. But once they do add cookies (ie: making calls with `withCredentials: true` set in their axios configuration), this CORS middleware will block it because we're not currently telling the middleware which websites it should allow cookies from.

It would be good to have that set up ahead of time at least for local development:

```suggestion
app.use(cors({ origin: ["http://localhost:3000", credentials: true }));
```

or with regex:
```suggestion
app.use(cors({ origin: [/localhost:3000$/], credentials: true }));
```

Later once we deploy these repos to Render.com, then we'll also need to add the deployed URL but we can do that when the time comes.

Reply:
Corrected commit- 46b69103aad60abd5f75962aba94b9fce32e31a3